subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c

```diff
@@ -1650,13 +1659,17 @@ int nrf_cloud_modem_info_json_encode(const struct nrf_cloud_modem_info *const mo
 		/* No modem info provided, use local */
 		err = get_modem_info();
 		if (err < 0) {
-			LOG_ERR("modem_info_params_get() failed: %d", err);
+			LOG_ERR("get_modem_info() failed: %d", err);
 			goto cleanup;
 		}
+		(void)k_mutex_lock(&modem_inf_mutex, K_FOREVER);
 		mpi = &modem_inf;
 	}
 
 	err = encode_modem_info_json_object(mpi, tmp, mod_inf->application_version);
+	if (mpi == &modem_inf) {
```

Comment:
can `mod_inf->mpi` definitely never already be equal to `&modem_inf`?

Maybe a dedicated boolean that gets set true prior to the lock would be a good idea just to sidestep even having to answer that question

Reply:
> would be a good idea just to sidestep even having to answer that question

🤔 
ok... flag added, discussion side-stepped.